modules/apps/foundation/user-associated-data/user-associated-data-api/src/main/java/com/liferay/user/associated/data/model/UADEntityAggregator.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.user.associated.data.model;
+
+import aQute.bnd.annotation.ProviderType;
+
+import java.util.List;
+
+/**
+ * @author William Newbury
+ */
+@ProviderType
+public interface UADEntityAggregator {
+
+	public long count(long userId);
+
+	public List<UADEntity> getUADEntities(long userId);
+
+	public UADEntity getUADEntity(String uadRegistryKey);
```

Comment:
It looks like the parameter here should be `uadEntityId` instead of `uadRegistryKey`.

Reply:
updated with those fixes